lib/AST/DiagnosticEngine.cpp

```diff
@@ -176,9 +176,35 @@ InFlightDiagnostic &InFlightDiagnostic::fixItReplace(SourceRange R,
     return fixItRemove(R);
 
   assert(IsActive && "Cannot modify an inactive diagnostic");
-  if (Engine && R.isValid())
-    Engine->getActiveDiagnostic().addFixIt(
-        Diagnostic::FixIt(toCharSourceRange(Engine->SourceMgr, R), Str));
+  if (R.isInvalid() || !Engine) return *this;
+
+  auto &SM = Engine->SourceMgr;
+  auto charRange = toCharSourceRange(Engine->SourceMgr, R);
+
+  // If we're replacing with something that wants spaces around it, do a bit of
+  // extra work so that we don't suggest extra spaces.
+  if (Str.back() == ' ') {
```

Comment:
@DougGregor 
It's redundant because [a few lines above](https://github.com/rintaro/swift/blob/ba982bc01cffaf52b27f0d7831df59045efd175b/lib/AST/DiagnosticEngine.cpp#L175-L176):
```cpp
   if (Str.empty())
      return fixItRemove(R);
```